connector_prestashop/models/res_partner/importer.py

```diff
@@ -0,0 +1,215 @@
+        groups = self.prestashop_record.get('associations', {}) \
+            .get('groups', {}).get(
+            self.backend_record.get_version_ps_key('group'), [])
+        if not isinstance(groups, list):
+            groups = [groups]
+        for group in groups:
+            self._import_dependency(group['id'],
+                                    'prestashop.res.partner.category')
+
+    def _after_import(self, binding):
+        super(ResPartnerImporter, self)._after_import(binding)
+        binder = self.binder_for()
+        ps_id = binder.to_external(binding)
+        self.env['prestashop.address'].with_delay(priority=10).import_batch(
+            backend=self.backend_record,
+            filters={'filter[id_customer]': '%d' % (ps_id,)})
+
+
+class PartnerBatchImporter(Component):
+    _name = 'prestashop.res.partner.batch.importer'
+    _inherit = 'prestashop.delayed.batch.importer'
+    _apply_on = 'prestashop.res.partner'
+
+
+class AddressImportMapper(Component):
+    _name = 'prestashop.address.mappper'
+    _inherit = 'prestashop.import.mapper'
+    _apply_on = 'prestashop.address'
+
+    direct = [
+        ('address1', 'street'),
+        ('address2', 'street2'),
+        ('city', 'city'),
+        ('other', 'comment'),
+        ('phone', 'phone'),
+        ('phone_mobile', 'mobile'),
+        ('postcode', 'zip'),
+        ('date_add', 'date_add'),
+        ('date_upd', 'date_upd'),
+        ('alias', 'alias'),
+        ('company', 'company'),
+        (external_to_m2o('id_customer'), 'prestashop_partner_id'),
+    ]
+
+    @mapping
+    def backend_id(self, record):
+        return {'backend_id': self.backend_record.id}
+
+    @mapping
+    def parent_id(self, record):
+        binder = self.binder_for('prestashop.res.partner')
+        parent = binder.to_internal(record['id_customer'], unwrap=True)
+        return {'parent_id': parent.id}
+
+    @mapping
+    def name(self, record):
+        parts = [record['firstname'], record['lastname']]
+        name = ' '.join(p.strip() for p in parts if p.strip())
+        return {'name': name}
+
+    @mapping
+    def customer(self, record):
+        return {'customer': True}
+
+    @mapping
+    def country(self, record):
+        if record.get('id_country'):
+            binder = self.binder_for('prestashop.res.country')
+            country = binder.to_internal(record['id_country'], unwrap=True)
+            return {'country_id': country.id}
+        return {}
+
+    @mapping
+    def company_id(self, record):
+        return {'company_id': self.backend_record.company_id.id}
+
+    @only_create
+    @mapping
+    def type(self, record):
+        # do not set 'contact', otherwise the address fields are shared with
+        # the parent
+        return {'type': 'other'}
+
+
+class AddressImporter(Component):
+    _name = 'prestashop.address.importer'
+    _inherit = 'prestashop.importer'
+    _apply_on = 'prestashop.address'
+
+    def _check_vat(self, vat):
```

Comment:
Maybe this function can receive the record country, and use it if the vat_number comes without the country code, like in the res.partner check_vat function.
```
def _check_vat(self, vat_number, partner_country):
        vat_country, vat_number_ = self.env["res.partner"]._split_vat(
            vat_number
        )
        if not self.env["res.partner"].simple_vat_check(vat_country, vat_number_):
            # if fails, check with country code from country
            country_code = partner_country.code
            if country_code:
                if not self.env["res.partner"].simple_vat_check(country_code.lower(), vat_number):
                    return False
        return True
```
This will reduce the number of partners who need to be checked